test/hotspot/gtest/nmt/test_nmt_malloclimit.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2022 SAP SE. All rights reserved.
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include "precompiled.hpp"
+#include "memory/allocation.hpp"
+#include "runtime/os.hpp"
+#include "services/mallocLimit.hpp"
+#include "services/memTracker.hpp"
+#include "services/nmtCommon.hpp"
+#include "utilities/debug.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/ostream.hpp"
+
+#include "testutils.hpp"
+#include "unittest.hpp"
+
+// Tests here just test the MallocLimit option parser. They are complemented
+// by more extensive jtreg tests (runtime/NMT/TestMallocLimit.java)
+static bool compare_limits(const malloclimit_t* a, const malloclimit_t* b) {
+  return a->v == b->v && a->f == b->f;
+}
+
+static bool compare_sets(const MallocLimitSet* a, const MallocLimitSet* b) {
+  if (compare_limits(a->global_limit(), b->global_limit())) {
+    for (int i = 0; i < mt_number_of_types; i++) {
+      if (!compare_limits(a->category_limit(NMTUtil::index_to_flag(i)),
+                          b->category_limit(NMTUtil::index_to_flag(i)))) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+static void test(const char* s, const MallocLimitSet& expected) {
+//  LOG_HERE("%s", s);
+  MallocLimitSet set;
+  const char* err;
+  ASSERT_TRUE(set.parse_malloclimit_option(s, &err)) << err;
+  ASSERT_TRUE(compare_sets(&set, &expected));
+}
+
+TEST(NMT, malloclimit_global) {
+  MallocLimitSet expected;
+
+  expected.set_global_limit(1 * G, malloclimit_mode_t::trigger_fatal);
+  test("1g", expected);
+  test("1024m", expected);
+  test("1048576k", expected);
+  test("1073741824", expected);
+
+  // Fatal is default, but can be specified explicitely
+  test("1g:fatal", expected);
+
+  expected.set_global_limit(2 * M, malloclimit_mode_t::trigger_oom);
+  test("2m:oom", expected);
+  test("2m:OOM", expected);
+  test("2048k:oom", expected);
+}
+
+TEST(NMT, malloclimit_per_category) {
```

Comment:
>malloclimit_per_category

Can we use CamelCase for test case names instead? This is a convention that we *should* use for gtest: http://google.github.io/googletest/faq.html

Unfortunately, we don't consistently do this, but I do want to avoid introducing new instances of this.

Reply:
I can see the point, though my names often follow a vague hierarchy if the suitename is already very generic. E.g. "malloclimit_xxx". - suitename is "NMT" and should remain so, all tests in this file concern MallocLimit, and then something. I like to separate "MallocLimit" visually from the "something".

I guess its okay, just a minor annoyance.